services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -0,0 +1,250 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import logging
+from pathlib import Path
+from typing import List, Optional, Set
+
+import duckdb
+from huggingface_hub._commit_api import (
+    CommitOperation,
+    CommitOperationAdd,
+    CommitOperationDelete,
+)
+from huggingface_hub.hf_api import HfApi
+from huggingface_hub.utils._errors import RepositoryNotFoundError
+from libcommon.config import DuckDbIndexConfig
+from libcommon.constants import PROCESSING_STEP_SPLIT_DUCKDB_INDEX_VERSION
+from libcommon.exceptions import (
+    DatasetNotFoundError,
+    NoIndexableColumnsError,
+    NotAvailableIndexFileError,
+    ParquetResponseEmptyError,
+    PreviousStepFormatError,
+    SplitNotFoundError,
+    SplitWithTooBigParquetError,
+)
+from libcommon.processing_graph import ProcessingStep
+from libcommon.simple_cache import get_previous_step_or_raise
+from libcommon.storage import StrPath
+from libcommon.utils import JobInfo, SplitHubFile
+
+from worker.config import AppConfig
+from worker.job_runners.split.split_job_runner import SplitCachedJobRunner
+from worker.utils import CompleteJobResult, hf_hub_url
+
+DATASET_TYPE = "dataset"
+STRING_FEATURE_DTYPE = "string"
+VALUE_FEATURE_TYPE = "Value"
+DUCKDB_DEFAULT_INDEX_FILENAME = "duckdb_index.db"
+CREATE_SEQUENCE_COMMAND = "CREATE OR REPLACE SEQUENCE serial START 1;"
+CREATE_INDEX_COMMAND = "PRAGMA create_fts_index('data', '__id', '*', overwrite=1);"
+CREATE_TABLE_COMMAND = "CREATE OR REPLACE TABLE data AS SELECT nextval('serial') AS __id, * FROM"
+# TODO: What if __id field already exist?
```

Comment:
maybe use something even less likely to exist, like `__hf_index_id`

Reply:
Good idea! I renamed it to __hf_index_id